test/nodes/test_connector.py

```diff
@@ -1,121 +1,174 @@
-
-    partial_content_match: list = [
-        {
-            "url": "https://haystack.deepset.ai/overview/v1.1.0/intro",
-            "partial_content": [
-                "Haystack is an open-source framework ",
-                "for building search systems that work intelligently ",
-                "over large document collections.",
-                "Recent advances in NLP have enabled the application of ",
-                "question answering, retrieval and summarization ",
-                "to real world settings and Haystack is designed to be ",
-                "the bridge between research and industry.",
-            ],
-        },
-        {
-            "url": "https://haystack.deepset.ai/overview/v1.1.0/use-cases",
-            "partial_content": [
-                "Expect to see results that highlight",
-                "the very sentence that contains the answer to your question.",
-                "Thanks to the power of Transformer based language models,",
-                "results are chosen based on compatibility in meaning",
-                "rather than lexical overlap.",
-            ],
-        },
-    ]
+def test_crawler_output_file_structure(test_url, tmp_path):
+    crawler = Crawler(output_dir=tmp_path)
+    paths = crawler.crawl(urls=[test_url + "/index.html"], crawler_depth=0)
+    with open(paths[0].absolute(), "r") as doc_file:
+        data = json.load(doc_file)
+        assert "content" in data
+        assert "meta" in data
+        assert isinstance(data["content"], str)
+        assert len(data["content"].split()) > 2
 
-    crawler = Crawler(output_dir=tmp_dir)
-    for _dict in partial_content_match:
-        url: str = _dict["url"]
-        partial_content: list = _dict["partial_content"]
 
-        doc_path = crawler.crawl(urls=[url], crawler_depth=0)
-        assert len(doc_path) == 1
+def test_crawler_filter_urls(test_url, tmp_path):
+    crawler = Crawler(output_dir=tmp_path)
+
+    print(crawler.crawl(urls=[test_url + "/index.html"], filter_urls=["page1"], crawler_depth=1))
+    assert len(crawler.crawl(urls=[test_url + "/index.html"], filter_urls=["page1"], crawler_depth=1)) == 1
+    assert not crawler.crawl(urls=[test_url + "/index.html"], filter_urls=["page3"], crawler_depth=1)
+    assert not crawler.crawl(urls=[test_url + "/index.html"], filter_urls=["google\.com"], crawler_depth=1)
 
-        for json_file in doc_path:
-            assert isinstance(json_file, Path)
-            with open(json_file.absolute(), "r") as read_file:
-                content = json.load(read_file)
-                assert isinstance(content["content"], str)
-                for partial_line in partial_content:
-                    assert search(partial_line, content["content"])
-                    assert partial_line in content["content"]
 
+def test_crawler_content(test_url, tmp_path):
+    expected_results = [
+        {"url": test_url + "/index.html", "partial_content": "test page content"},
```

Comment:
Nit: I usually struggle reading table-driven tests, I think it would help a lot using different naming, like:
```python
    test_cases = [
        {"input": test_url + "/index.html", "expected": "test page content"},
        {"input": test_url + "/page1.html", "expected": "test page 1 content"},
        {"input": test_url + "/page2.html", "expected": "test page 2 content"},
    ]

    crawler = Crawler(output_dir=tmp_path)
    for test_case in test_cases:
        paths = crawler.crawl(urls=[test_case["input"]], crawler_depth=0)
        with open(paths[0].absolute(), "r") as read_file:
            content = json.load(read_file)
            assert test_case["expected"] in content["content"]
```

Reply:
I think this test can simply be removed. I added content check to many other tests to make sure the content is still checked somewhere.